src/databaseService/index.js

```diff
@@ -0,0 +1,59 @@
+import 'react-native'
+import * as firebase from 'firebase'
+
+const set = async (refPath, data) => {
+  try {
+    await firebase.database().ref(refPath).set(data)
+  } catch (error) {
+    console.log(error.message)
+  }
+}
+
+const get = async (refPath, mode = 'value') => {
+  let values = null
+  try {
+    const databaseSnapshot = await firebase.database().ref(refPath).once(mode)
+    values = databaseSnapshot.val()
+  } catch (error) {
+    console.log(error.message)
+  }
+  return values
+}
+
+const update = async (refPath, data) => {
+  try {
+    await firebase.database().ref(refPath).update(data)
+  } catch (error) {
+    console.log(error.message)
+  }
+}
+
+const getEmailFromJson = (json) => {
+  return json[Object.keys(json)[0]].email
+}
+
+const queryEmailFromUsername = async (username, refPath = 'users/') => {
+  const ref = firebase.database().ref(refPath)
+  let email = null
+  try {
+    const databaseSnapshot = await ref.orderByChild('username').equalTo(username).once('value')
+    const json = databaseSnapshot.val()
+    email = json && getEmailFromJson(json)
+  } catch (error) {
+    console.log(error.message)
+  }
+  return email
+}
+
+const usernameAlreadyInUse = async (username, refPath = 'users/') => {
+  const email = await queryEmailFromUsername(username, refPath)
+  return email !== null
```

Comment:
Let's leave a blank line before return statements (when they are among other statements within the same block).